src/lib/data/alignment.js

```diff
@@ -63,57 +63,110 @@ export default class Alignment {
   }
 
   shouldFinishAlignmentGroup (token) {
-    return this.tokenInUnfinishedGroup(token) && this.tokenTheSameTextTypeAsStart(token)
+    return this.tokenInActiveGroup(token) && this.tokenTheSameTextTypeAsStart(token)
+  }
+
+  shouldBeRemovedFromAlignmentGroup (token) {
+    return this.tokenInActiveGroup(token) && !this.tokenTheSameTextTypeAsStart(token)
   }
 
   shouldStartNewAlignmentGroup (token) {
-    return !this.currentAlignmentGroup
+    return !this.activeAlignmentGroup
   }
 
   startNewAlignmentGroup (token) {
-    this.currentAlignmentGroup = new AlignmentGroup(token)
+    this.activeAlignmentGroup = new AlignmentGroup(token)
   }
 
   addToAlignmentGroup (token) {
-    if (this.currentAlignmentGroup[token.textType]) {
-      this.currentAlignmentGroup.add(token)
+    if (this.activeAlignmentGroup[token.textType]) {
+      this.activeAlignmentGroup.add(token)
+    } else {
+      console.error(this.l10n.getMsg('ALIGNMENT_ERROR_ADD_TO_ALIGNMENT'))
+    }
+  }
+
+  removeFromAlignmentGroup (token) {
+    if (this.activeAlignmentGroup[token.textType]) {
+      this.activeAlignmentGroup.remove(token)
+      this.removeFromAlignmentIds(token.idWord)
     } else {
       console.error(this.l10n.getMsg('ALIGNMENT_ERROR_ADD_TO_ALIGNMENT'))

```

Comment:
very minor point, but the error on line 94 should probably be specific to removing not adding a token

Reply:
oh, thank you for the catch